src/visual.ts

```diff
@@ -218,35 +218,28 @@ export class PowerKPI implements IVisual {
             secondTooltipVariance
         ]
         settingsToFilterByAxis.forEach(card => {
-            const newSlices = [...card.slices]
-            this.addArrayItems(newSlices, [card.displayUnits, card.precision, card.format])
-            if (card.shouldNumericPropertiesBeHiddenByType
-                && this.axisType !== DataRepresentationTypeEnum.NumberType
-            ) {
-                this.removeArrayItem(newSlices, card.displayUnits)
-                this.removeArrayItem(newSlices, card.precision)
-            }
-    
-            if (this.axisType !== DataRepresentationTypeEnum.NumberType
-                && this.axisType !== DataRepresentationTypeEnum.DateType
-            ) {
-                this.removeArrayItem(newSlices, card.format)
-            }
-            card.slices = newSlices
+            const shouldNumericPropertiesBeHidden = 
```

Comment:
setting filtering is not related to visual logic, this should be implemented in formatting model